db-service/lib/deep-queries.js

```diff
@@ -205,16 +209,17 @@ const _hasManagedElements = target => {
 /**
  * @param {unknown[]} diff
  * @param {import('@sap/cds/apis/csn').Definition} target
+ * @param {Map<String, Object>} deletes
```

Comment:
why is `updates` not a map?:)
```suggestion
 * @param {Map<String, Object>} deletes
 * @param {Map<String, Object>} inserts
 * @param {Object[]} updates
```

Reply:
Yeah, it's a bit inconsistent, but for updates, I can't group by target.name :(

Reply:
and constructing an identifier for each key combination seems a bit excessive :)